float32 deg1
float32 deg2
float32 deg3
